src/management-system-v2/lib/data/legacy/folders.ts

```diff
@@ -0,0 +1,225 @@
+        throw new Error(`Environment ${folder.environmentId} Multiple root folders`);
+
+      foldersMetaObject.rootFolders[folder.environmentId] = folder;
+    }
+
+    foldersMetaObject.folders[folder.id] = { folder, children: [] };
+  }
+
+  //populate children
+  //children fill their parent's children array
+  for (const folder of storedFolders) {
+    // skip roots
+    if (!folder.parentId) continue;
+
+    const parentData = foldersMetaObject.folders[folder.parentId];
+    if (!parentData)
+      throw new Error(`Inconsistency in folder structure: folder ${folder.id} has no parent`);
+
+    parentData.children.push(folder);
+  }
+}
+init();
+
+export function getRootFolder(environmentId: string, ability?: Ability) {
+  const rootFolder = foldersMetaObject.rootFolders[environmentId];
+  if (!rootFolder) throw new Error(`MS Error: environment ${environmentId} has no root folder`);
+
+  if (ability && !ability.can('view', toCaslResource('Folder', rootFolder)))
+    throw new Error('Permission denied');
+
+  return rootFolder;
+}
+
+export function getFolderById(folderId: string, ability?: Ability) {
+  const folderData = foldersMetaObject.folders[folderId];
+  if (!folderData) throw new Error('Folder not found');
+
+  if (ability && !ability.can('view', toCaslResource('Folder', folderData.folder)))
+    throw new Error('Permission denied');
+
+  return folderData.folder;
+}
+
+export function getFolderChildren(folderId: string, ability?: Ability) {
+  const folderData = foldersMetaObject.folders[folderId];
+  if (!folderData) throw new Error('Folder not found');
+
+  if (ability && !ability.can('view', toCaslResource('Folder', folderData.folder)))
+    throw new Error('Permission denied');
+
+  return folderData.children;
+}
+
+export function createFolder(folderInput: z.infer<typeof FolderSchema>, ability?: Ability) {
+  const folder = FolderSchema.parse(folderInput);
+  if (!folder.id) folder.id = v4();
+
+  // Checks
+  if (ability && !ability.can('create', toCaslResource('Folder', folder)))
+    throw new Error('Permission denied');
+
+  if (foldersMetaObject.folders[folder.id]) throw new Error('Folder already exists');
+
+  const parentFolderData = folder.parentId && foldersMetaObject.folders[folder.parentId];
+  if (folder.parentId) {
+    if (!parentFolderData) throw new Error('Parent folder does not exist');
+
+    if (parentFolderData.folder.environmentId !== folder.environmentId)
+      throw new Error('Parent folder is in a different environment');
+  } else {
+    if (foldersMetaObject.rootFolders[folder.environmentId])
+      throw new Error(`Environment ${folder.environmentId} already has a root folder`);
+  }
+
+  // Store
+  const newFolder = {
+    ...folder,
+    createdAt: new Date().toISOString(),
+    updatedAt: new Date().toISOString(),
+  } as Folder;
+
+  foldersMetaObject.folders[folder.id] = { folder: newFolder, children: [] };
+
+  if (parentFolderData) parentFolderData.children.push(newFolder);
```

Comment:
Could changing the children of a folder be considered an update that warrants an update to the `updatedAt` property of that folder?

Reply:
Yes, I also updated it in other functions where it should be done as well